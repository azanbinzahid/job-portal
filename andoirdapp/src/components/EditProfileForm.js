import React from 'react';
import {connect} from 'react-redux';
import {useFormik} from 'formik';
import * as Yup from 'yup';
import {editUser} from '../redux/actions';
import {Form, Input, Item, Text, Button, Label, Textarea} from 'native-base';

const validationSchema = Yup.object().shape({
  email: Yup.string().email().required(),
  firstName: Yup.string().max(16).required(),
  lastName: Yup.string().max(16).required(),
  profile: Yup.object().shape({
    bio: Yup.string().max(500).required(),
    location: Yup.string().max(30).required(),
    education: Yup.string().max(300).required(),
    experiance: Yup.string().max(300).required(),
    // birthDate: Yup.date().required(),
  }),
});

const EditProfileForm = (props) => {
  const {handleSubmit, handleChange, handleBlur, values, errors} = useFormik({
    initialValues: props.userDetails,
    validationSchema,
    onSubmit(values) {
      props.editUser(values);
    },
  });

  return (
    <Form>
      <Item floatingLabel>
        <Label> Email </Label>
        <Input
          placeholder="email"
          value={values.email}
          onChangeText={handleChange('email')}
          onBlurText={handleBlur('email')}
        />
      </Item>
      <Text style={StyleSheet.text}>{errors.email}</Text>

      <Item floatingLabel>
        <Label> First Name </Label>
        <Input
          placeholder="firstName"
          value={values.firstName}
          onChangeText={handleChange('firstName')}
          onBlurText={handleBlur('firstName')}
        />
      </Item>
      <Text style={StyleSheet.text}>{errors.firstName}</Text>

      <Item floatingLabel>
        <Label> Last Name </Label>
        <Input
          placeholder="lastName"
          value={values.lastName}
          onChangeText={handleChange('lastName')}
          onBlurText={handleBlur('lastName')}
        />
      </Item>
      <Text style={StyleSheet.text}>{errors.lastName}</Text>

      <Item floatingLabel>
        <Label> Bio </Label>
        <Input
          multiline={true}
          numberOfLines={5}
          placeholder="bio"
          value={values.profile.bio}
          onChangeText={handleChange('profile.bio')}
          onBlurText={handleBlur('profile.bio')}
        />
      </Item>
      <Text style={StyleSheet.text}>{errors.profile && errors.bio}</Text>

      <Item floatingLabel>
        <Label> Location </Label>
        <Input
          placeholder="location"
          value={values.profile.location}
          onChangeText={handleChange('profile.location')}
          onBlurText={handleBlur('profile.location')}
        />
      </Item>
      <Text style={StyleSheet.text}>{errors.profile && errors.location}</Text>

      <Item floatingLabel>
        <Label> Education </Label>
        <Input
          multiline={true}
          numberOfLines={5}
          placeholder="education"
          value={values.profile.education}
          onChangeText={handleChange('profile.education')}
          onBlurText={handleBlur('profile.education')}
        />
      </Item>
      <Text style={StyleSheet.text}>{errors.profile && errors.lastName}</Text>

      <Item floatingLabel>
        <Label> Experiance </Label>
        <Input
          multiline={true}
          numberOfLines={5}
          placeholder="experiance"
          value={values.profile.experiance}
          onChangeText={handleChange('profile.experiance')}
          onBlurText={handleBlur('profile.experiance')}
        />
      </Item>
      <Text style={StyleSheet.text}>{errors.profile && errors.lastName}</Text>

      <Button bordered style={StyleSheet.button} onPress={handleSubmit}>
        <Text>Update</Text>
      </Button>
    </Form>
  );
};

const StyleSheet = {
  button: {
    alignSelf: 'center',
    margin: 15,
  },
  text: {
    alignSelf: 'center',
    textAlign: 'center',
  },
};

const mapStateToProps = (state) => ({
  userDetails: state.userReducer.user,
});

const mapDispatchToProps = (dispatch) => {
  return {
    editUser: (userInfo) => dispatch(editUser(userInfo)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(EditProfileForm);
