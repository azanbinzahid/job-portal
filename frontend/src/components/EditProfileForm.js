import React from 'react'
import {connect} from 'react-redux'
import { Form, Button, Jumbotron, Container } from 'react-bootstrap';
import { useFormik } from "formik";
import * as Yup from 'yup'
import {editUser} from 'redux/actions'

const validationSchema = Yup.object().shape({   
    email: Yup     
        .string()     
        .email()     
        .required(),
    firstName: Yup     
        .string()     
        .max(16)     
        .required(),   
    lastName: Yup     
        .string()     
        .max(16)     
        .required(), 
    
    // validation errors, need to fix 
    profile: Yup.object().shape({  
        bio: Yup     
            .string()     
            .max(500)     
            .required(),   
        location: Yup     
            .string()     
            .max(30)     
            .required(),   
        education: Yup     
            .string()     
            .max(300)     
            .required(),   
        experiance: Yup     
            .string()     
            .max(300)     
            .required(),   

        birthDate: Yup     
            .date()     
            .required()   
    })
})

const EditProfileForm = (props) => {
    const { handleSubmit, handleChange, values, errors} = useFormik({
        initialValues: props.userDetails,
        validationSchema,
        onSubmit(values) {
            props.editUser(values)
        }
    })


    return(
        <Container>
        <Jumbotron>
        <Form onSubmit={handleSubmit}>
            <Form.Group controlId="formBasicEmail">
            <Form.Label>Email address</Form.Label>
                <Form.Control 
                    type="email"
                    name="email"
                    placeholder="Email"
                    value={values.email}
                    onChange={handleChange}
                    />
            </Form.Group>
            {errors.email}
            <Form.Group controlId="formBasicFirstName">
            <Form.Label>First Name</Form.Label>
                <Form.Control 
                    type="text"
                    name="firstName"
                    placeholder="First Name"
                    value={values.firstName}
                    onChange={handleChange}
                />
            </Form.Group>
            {errors.firstName}

            <Form.Group controlId="formBasicLastName">
            <Form.Label>Last Name</Form.Label>
                <Form.Control 
                    type="text"
                    name="lastName"
                    placeholder="Last Name"
                    value={values.lastName}
                    onChange={handleChange}
                />
            </Form.Group>
            {errors.lastName}

            <Form.Group controlId="formBasicBio">
            <Form.Label>Bio</Form.Label>
                <Form.Control 
                    as="textarea"
                    rows="3"
                    name="profile.bio"
                    placeholder="Bio"
                    value={values.profile.bio}
                    onChange={handleChange}
                />
            </Form.Group>
            {errors.bio}

            <Form.Group controlId="formBasicLocation">
            <Form.Label>Current Location</Form.Label>
                <Form.Control 
                    type="text"
                    name="profile.location"
                    placeholder="Location"
                    value={values.profile.location}
                    onChange={handleChange}
                />
            </Form.Group>
            {errors.location}

            <Form.Group controlId="formBasicEducation">

            <Form.Label>Education</Form.Label>
                <Form.Control 
                    as="textarea"
                    rows="3"
                    name="profile.education"
                    placeholder="Education"
                    value={values.profile.education}
                    onChange={handleChange}
                />
            </Form.Group>
            {errors.education}

            <Form.Group controlId="formBasicEducation">
            <Form.Label>Experiance</Form.Label>
                <Form.Control 
                    as="textarea"
                    rows="3"
                    name="profile.experiance"
                    placeholder="Experiance"
                    value={values.profile.experiance}
                    onChange={handleChange}
                />
            </Form.Group>
            {errors.experiance}


            <Form.Group controlId="formBasicDate">
            <Form.Label>Birthdate</Form.Label>

                <Form.Control 
                    type="date"
                    name="profile.date"
                    placeholder="Birth Day"
                    value={values.profile.birthDate}
                    onChange={handleChange}
                />
            </Form.Group>
            {errors.date}


            <Button variant="primary" type="submit">
                Update Profile
            </Button>
        </Form>
        </Jumbotron>
        </Container>
    )
}


const mapStateToProps = state => ({
    userDetails: state.userReducer.user
  });

const mapDispatchToProps = (dispatch) => {
    return {
        editUser: (userInfo) => dispatch(editUser(userInfo))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditProfileForm);
