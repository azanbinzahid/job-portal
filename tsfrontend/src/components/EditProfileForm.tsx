import React, { FC } from "react";
import { connect } from "react-redux";
import { Form, Button, Jumbotron, Container } from "react-bootstrap";
import { useFormik } from "formik";
import * as Yup from "yup";
import { editUser } from "redux/actions";
import { RootState } from "redux/reducers";
import { User } from "redux/types";
import TextField from "./TextField";

const validationSchema = Yup.object().shape({
  email: Yup.string().email().required(),
  firstName: Yup.string().max(16).required(),
  lastName: Yup.string().max(16).required(),

  // validation errors, need to fix
  profile: Yup.object().shape({
    bio: Yup.string().max(500).required(),
    location: Yup.string().max(30).required(),
    education: Yup.string().max(300).required(),
    experiance: Yup.string().max(300).required(),

    birthDate: Yup.date().required(),
  }),
});

type Props = {
  userDetails: User;
  editUser: (values: User) => void;
};
const EditProfileForm: FC<Props> = (props) => {
  const { handleSubmit, handleChange, values, errors } = useFormik({
    initialValues: props.userDetails,
    validationSchema,
    onSubmit(values) {
      props.editUser(values);
    },
  });

  return (
    <Container>
      <Jumbotron>
        <Form
          onSubmit={(e: React.FormEvent<HTMLFormElement>) => handleSubmit(e)}
        >
          <TextField
            label="Email Address"
            type="email"
            name="email"
            placeholder="Email"
            value={values.email}
            onChange={handleChange}
            error={errors.email}
          />
          <TextField
            label="First Name"
            type="text"
            name="firstName"
            placeholder="First Name"
            value={values.firstName}
            onChange={handleChange}
            error={errors.firstName}
          />

          <TextField
            label="Last Name"
            type="text"
            name="lastName"
            placeholder="Last Name"
            value={values.lastName}
            onChange={handleChange}
            error={errors.lastName}
          />

          <Form.Group>
            <Form.Label>Bio</Form.Label>
            <Form.Control<React.ElementType>
              as="textarea"
              rows="3"
              name="profile.bio"
              placeholder="Bio"
              value={values.profile.bio}
              onChange={handleChange}
            />
          </Form.Group>
          {errors.profile && errors.profile.bio}

          <TextField
            label="Current Location"
            type="text"
            name="profile.location"
            placeholder="Location"
            value={values.profile.location}
            onChange={handleChange}
            error={errors.profile && errors.profile.location}
          />

          <Form.Group>
            <Form.Label>Education</Form.Label>
            <Form.Control<React.ElementType>
              as="textarea"
              rows="3"
              name="profile.education"
              placeholder="Education"
              value={values.profile.education}
              onChange={handleChange}
            />
          </Form.Group>
          {errors.profile && errors.profile.education}

          <Form.Label>Experiance</Form.Label>
          <Form.Control<React.ElementType>
            as="textarea"
            rows="3"
            name="profile.experiance"
            placeholder="Experiance"
            value={values.profile.experiance}
            onChange={handleChange}
          />
          {errors.profile && errors.profile.experiance}

          <Form.Group>
            <Form.Label>Birthdate</Form.Label>
            <Form.Control
              type="date"
              name="profile.birthDate"
              placeholder="Birth Day"
              value={new Date(values.profile.birthDate).toLocaleDateString()}
              onChange={handleChange}
            />
          </Form.Group>
          {errors.profile && errors.profile.birthDate}

          <Button variant="primary" type="submit">
            Update Profile
          </Button>
        </Form>
      </Jumbotron>
    </Container>
  );
};

const mapStateToProps = (state: RootState) => ({
  userDetails: state.userReducer.user,
});

const mapDispatchToProps = {
  editUser,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditProfileForm as any);
