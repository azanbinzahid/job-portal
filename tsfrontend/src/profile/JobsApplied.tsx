import React, { FC } from "react";
import { ListGroup, ListGroupItem } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";
import { Job } from "redux/types";

type Props = {
  jobsApplied: Job[];
};

const JobsApplied: FC<Props> = ({ jobsApplied }) => {
  return (
    <ListGroup>
      <h3> Jobs Applied </h3>
      {jobsApplied.map((job, index) => (
        <LinkContainer key={index} to={`/jobs/${job.id}`}>
          <ListGroupItem action key={index}>
            {job.title}
          </ListGroupItem>
        </LinkContainer>
      ))}
    </ListGroup>
  );
};

export default JobsApplied;
