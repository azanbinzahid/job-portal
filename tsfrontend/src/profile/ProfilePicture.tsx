import React, { FC } from "react";
import { connect } from "react-redux";
import { Image, Form, Button, Container } from "react-bootstrap";
import { useFormik } from "formik";
import { uploadImage } from "redux/actions";
import { RootState } from "redux/reducers";

type Props = {
  profileImage: string;
  uploadImage: (values: { image: string }) => void;
};
const ProfilePicture: FC<Props> = (props) => {
  const { handleSubmit, setFieldValue } = useFormik({
    initialValues: { image: "" },
    onSubmit(values) {
      props.uploadImage(values);
    },
  });

  return (
    <Container<React.ElementType> align="center" className="pb-5">
      <Image src={process.env.REACT_APP_BASE_URL + props.profileImage} fluid />
      <Form
        inline
        onSubmit={(e: React.FormEvent<HTMLFormElement>) => handleSubmit(e)}
      >
        <Form.File<React.ElementType>
          name="image"
          onChange={(event: any) => {
            setFieldValue("image", event.currentTarget.files[0]);
          }}
        />

        <Button variant="primary" type="submit">
          Update Image
        </Button>
      </Form>
    </Container>
  );
};

const mapStateToProps = (state: RootState) => ({
  profileImage: state.userReducer.user.profile.image,
});

const mapDispatchToProps = {
  uploadImage,
};

export default connect(mapStateToProps, mapDispatchToProps)(ProfilePicture);
