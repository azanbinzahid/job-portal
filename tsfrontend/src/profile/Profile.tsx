import React, { FC } from "react";
import { connect } from "react-redux";
import { Container, Row, Col } from "react-bootstrap";
import EditProfileForm from "profile/EditProfileForm";
import ProfilePicture from "profile/ProfilePicture";
import JobsApplied from "profile/JobsApplied";
import { RootState } from "redux/reducers";
import { Job } from "redux/types";
import SectionHeading from "app/common/SectionHeading";

type Props = {
  firstName: string;
  lastName: string;
  profileImage: string;
  jobsApplied: Job[];
};
export const Profile: FC<Props> = (props) => {
  return (
    <Container>
      <SectionHeading
        title={props.firstName + " " + props.lastName}
        subtitle="Profile Page"
      />
      <Row>
        <Col md="7">
          <EditProfileForm />
        </Col>
        <Col md="5">
          <ProfilePicture />
          <JobsApplied jobsApplied={props.jobsApplied} />
        </Col>
      </Row>
    </Container>
  );
};

const mapStateToProps = (state: RootState) => ({
  firstName: state.userReducer.user.firstName,
  lastName: state.userReducer.user.lastName,
  profileImage: state.userReducer.user.profile.image,
  jobsApplied: state.userReducer.user.jobsApplied,
});

export default connect(mapStateToProps)(Profile);
