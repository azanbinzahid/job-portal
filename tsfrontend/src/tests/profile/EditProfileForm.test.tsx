import React from "react";
import { shallow } from "enzyme";
import { EditProfileForm } from "profile/EditProfileForm";

const job = {
  id: 1,
  location: ["Lahore"],
  category: ["Customer Service"],
  qualification: ["Bachelors"],
  applicants: ["admin"],
  company: "Modern Tech",
  title: "WordPress Developer",
  salaray: 60000,
  description: "This is test job description",
  experiance: 1,
  datestamp: new Date(11 / 11 / 11),
};

const profile = {
  location: "",
  bio: "",
  education: "",
  experiance: "",
  birthDate: new Date(11 / 11 / 11),
  image: "",
};

const user = {
  username: "",
  firstName: "",
  lastName: "",
  email: "",
  jobsApplied: [job, job],
  profile: profile,
};

test("renders the component", () => {
  const component = shallow(
    <EditProfileForm userDetails={user} editUser={jest.fn()} />
  );
  expect(component).toMatchSnapshot();
});
