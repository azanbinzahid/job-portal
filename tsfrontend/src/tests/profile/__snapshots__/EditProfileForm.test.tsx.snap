// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Container
  fluid={false}
>
  <Jumbotron
    fluid={false}
  >
    <Form
      inline={false}
      onSubmit={[Function]}
    >
      <TextField
        label="Email Address"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
      <TextField
        label="First Name"
        name="firstName"
        onChange={[Function]}
        placeholder="First Name"
        type="text"
        value=""
      />
      <TextField
        label="Last Name"
        name="lastName"
        onChange={[Function]}
        placeholder="Last Name"
        type="text"
        value=""
      />
      <FormGroup>
        <FormLabel
          column={false}
          srOnly={false}
        >
          Bio
        </FormLabel>
        <FormControl
          as="textarea"
          name="profile.bio"
          onChange={[Function]}
          placeholder="Bio"
          rows="3"
          value=""
        />
      </FormGroup>
      <TextField
        label="Current Location"
        name="profile.location"
        onChange={[Function]}
        placeholder="Location"
        type="text"
        value=""
      />
      <FormGroup>
        <FormLabel
          column={false}
          srOnly={false}
        >
          Education
        </FormLabel>
        <FormControl
          as="textarea"
          name="profile.education"
          onChange={[Function]}
          placeholder="Education"
          rows="3"
          value=""
        />
      </FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Experiance
      </FormLabel>
      <FormControl
        as="textarea"
        name="profile.experiance"
        onChange={[Function]}
        placeholder="Experiance"
        rows="3"
        value=""
      />
      <FormGroup>
        <FormLabel
          column={false}
          srOnly={false}
        >
          Birthdate
        </FormLabel>
        <FormControl
          name="profile.birthDate"
          onChange={[Function]}
          placeholder="Birth Day"
          type="date"
          value="1/1/1970"
        />
      </FormGroup>
      <Button
        active={false}
        disabled={false}
        type="submit"
        variant="primary"
      >
        Update Profile
      </Button>
    </Form>
  </Jumbotron>
</Container>
`;
